packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -686,6 +687,53 @@ describe('TransactionController', () => {
       );
     });
 
+    it.each([
+      [
+        'adds single unapproved transactions with same actionId',
+        ACTION_ID_MOCK,
+        ACTION_ID_MOCK,
+        1,
+      ],
+      [
+        'adds multiple unapproved transactions with different actionId',
+        ACTION_ID_MOCK,
+        '00000',
+        2,
+      ],
+    ])(
+      '%s',
+      async (_, firstActionId, secondActionId, expectedTransactionCount) => {
+        const controller = newController();
+
+        const mockOrigin = 'origin';
+
+        await controller.addTransaction(
+          {
+            from: ACCOUNT_MOCK,
+            to: ACCOUNT_MOCK,
+          },
+          {
+            origin: mockOrigin,
+            actionId: firstActionId,
+          },
+        );
+
+        await controller.addTransaction(
```

Comment:
Following my action about skipping approvals and waiting for the event, we'll need another test to check that the duplicate transaction result doesn't resolve until the first transaction result resolves.

Reply:
Included this additional test to ensure the duplicate transaction result does not resolve until the first transaction.